go.mod

```diff
@@ -10,6 +10,7 @@ require (
 	github.com/go-sql-driver/mysql v1.5.0
 	github.com/golang/mock v1.6.0
 	github.com/golang/protobuf v1.5.2
+	github.com/google/go-cmp v0.5.9
```

Comment:
I'd like to use `go-cmp` since comparing the expected struct and the actual one (eyeball grep) is too harder...
Actually, I had a hard time fixing tests in this PR.
WDYT?

Reply:
I see, I think, if that is easier to use this lib instead of reflect , we might to want re-use it everywhere across our unit tests in the future.
Maybe make it as best practice to write unit test in our [Developer Guide](https://github.com/kubeflow/katib/blob/master/docs/developer-guide.md).

What are your thoughts @tenzen-y  @johnugeorge ?

Reply:
SGTM

Reply:
I'd like to work on updating the developer guide.